main.tf

```diff
@@ -97,10 +97,11 @@ module "eks" {
   cluster_endpoint_public_access       = var.cluster_api_mode == "public" ? true : false
   cluster_endpoint_public_access_cidrs = local.cluster_endpoint_public_access_cidrs
 
-  subnet_ids  = module.vpc.private_subnets
-  vpc_id      = module.vpc.vpc_id
-  tags        = local.tags
-  enable_irsa = var.autoscaling_enabled
+  control_plane_subnet_ids = module.vpc.private_subnets                                                           # AWS requires two or more subnets in different Availability Zones for your cluster's control plane.
+  subnet_ids               = var.enable_multi_zone ? module.vpc.private_subnets : [module.vpc.private_subnets[0]] # Specifies the list of subnets in which the worker nodes of the EKS cluster will be launched.
```

Comment:
In general I prefer comments above the code snippet its describing. But not a requirement ;)

Reply:
I agree with that change, and the rest of the code also follows that standard. 
Lines updated.